test/envoye2e/driver/envoy.go

```diff
@@ -233,7 +233,7 @@ func downloadEnvoy(ver string) (string, error) {
 	}
 	defer os.RemoveAll("usr/")
 
-	cpCmd := exec.Command("cp", src, dst)
+	cpCmd := exec.Command("cp", "-n", src, dst)
```

Comment:
Shouldn't this check be performed before we download Envoy? It's a bit wasteful to download all those bytes if we're not going to use the binary.

Reply:
There is already a check at line 209, but duplicate download still happens some time due to test runs parallel.

Reply:
Ah, OK.